### PR TITLE
autobuild for debian and redhat

### PR DESCRIPTION
**Pull request by smusali**
- add jobs to automate building packages for `debian` and `redhat`
- restructure `linux`-specific scripts
- update `workflow`
- update `.gitignore`

these changes are not for pushing/deploying/publishing/releasing anything - this is just for building the packages

`fpm` steps are taken from [here](https://github.com/logdna/logdna-agent/blob/master/Gruntfile.js#L24-L25)

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -4,17 +4,103 @@ executors:
...
+              --license ${LICENSE}\
+              --vendor ${VENDOR}\
+              --description ${DESCRIPTION}\
+              --url ${URL}\
+              --maintainer ${MAINTAINER}\
+              --before-remove ./.build/scripts/before-remove\
+              --after-upgrade ./.build/scripts/after-upgrade\
+              --force --deb-no-default-config-files\
+                ./.build/logdna-agent=/usr/bin/logdna-agent\
+                ./.build/scripts/init-script=/etc/init.d/logdna-agent\
```
Comment (mindjiver) [inline .circleci/config.yml]: We don't ship a SystemD unit file?

File: .circleci/config.yml [hunk 2]
```diff
@@ -4,17 +4,103 @@ executors:
...
+              --license ${LICENSE}\
+              --vendor ${VENDOR}\
+              --description ${DESCRIPTION}\
+              --url ${URL}\
+              --maintainer ${MAINTAINER}\
+              --before-remove ./.build/scripts/before-remove\
+              --after-upgrade ./.build/scripts/after-upgrade\
+              --force\
+                ./.build/logdna-agent=/usr/bin/logdna-agent\
+                ./.build/scripts/init-script=/etc/init.d/logdna-agent\
```
Comment (mindjiver) [inline .circleci/config.yml]: Same here, only SysV init?

File: .circleci/config.yml (hunk 1 above)
Comment (smusali) [reply .circleci/config.yml]: no - idk why but it's been like this for always

File: .circleci/config.yml (hunk 2 above)
Comment (smusali) [reply .circleci/config.yml]: yea, just `init.d`

**Comment by smusali** [comment]
@RYadav95 @mindjiver any extra feedback?

**Comment by RYadav95** [comment]
Overall, everything looks good but I would like to know if you had any issues using two different docker images inside a single executor. Also, I think we can get rid of some code by using logdna orbs. Will submit a PR for that as soon as it's ready.

File: .circleci/config.yml (hunk 1 above)
Comment (mindjiver) [reply .circleci/config.yml]: I'll dig around and see if this is something we can add, I guess Centos6 still needs a SystemV file so. Anyways. 👍

**Comment by smusali** [comment]
> Overall, everything looks good but I would like to know if you had any issues using two different docker images inside a single executor. Also, I think we can get rid of some code by using logdna orbs. Will submit a PR for that as soon as it's ready.

You are right, @RYadav95! I will try to split it into 2 jobs

**Comment by smusali** [comment]
@RYadav95 didn't split but separated `ruby` dependency

**Pull request merged**